source/adapters/level_zero/device.cpp

```diff
@@ -1545,8 +1545,9 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetGlobalTimestamps(
 ) {
   const uint64_t &ZeTimerResolution =
       Device->ZeDeviceProperties->timerResolution;
+  auto ValidBits = Device->ZeDeviceProperties->kernelTimestampValidBits;
```

Comment:
when is it possible for this value to be > 64? Isn't that a bug in L0?
Shouldn't this be value capped at 64 when it exceeds it?

Reply:
Main intention of the PR was to fix == 64 case, I didn't get the case >64.  This implementation still caps it at 64 bit, right?  even though I believe you are right that L0 runtime caps it itself as well. Do you want me to change this it to "assert" for >64 bit case?

Reply:
in that case, I think this should become `max(ValidBits, 63)`. Maybe create a function in device that returns `TimestampMask` instead of repeating this 3x ?

Reply:
Thanks for suggestions, fixed.